zebra-consensus/src/verify/block.rs

```diff
@@ -492,36 +493,69 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    #[spandoc::spandoc]
+    async fn header_solution() -> Result<(), Report> {
+        zebra_test::init();
+
+        // Service variables
+        let state_service = Box::new(zebra_state::in_memory::init());
+        let mut block_verifier = super::init(state_service.clone());
+
+        let ready_verifier_service = block_verifier.ready_and().await.map_err(|e| eyre!(e))?;
+
+        // Get a valid block
+        let mut block =
+            Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_415000_BYTES[..])
+                .expect("block test vector should deserialize");
+
+        // This should be ok
+        ready_verifier_service
+            .call(Arc::new(block.clone()))
+            .await
+            .map_err(|e| eyre!(e))?;
+
+        // Change nonce to something invalid
+        block.header.nonce = [0; 32];
+
+        // Error: invalid equihash solution for BlockHeader
+        ready_verifier_service
```

Comment:
one of the constraints of `tower::Service` is that you must call `ready` on the service before each `call` or else implementations are permitted to panic. 

https://docs.rs/tower/0.3.1/tower/trait.Service.html#tymethod.call

To avoid this we should copy line 505 to right above this line so that we get a new instance of the ready service for the second call.

Reply:
edit, I have no idea what code this was on, it doesn't seem to show up in the diff (damn you github), so I'm gonna just mark this as reviewed and go ahead and merge.

Reply:
Right, i did this in `coinbase` test but missed it in `header_solution`. Thanks for pushing a fix here.